### PR TITLE
Fix warnings and errors reported by Sentry

### PR DESCRIPTION
**Pull request by mardelnet**
This PR fixes the following warnings and errors reported by Sentry:

- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/954
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/955
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/956
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/957
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/1885
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/109
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/160
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/202
- https://sentry.greenpeace.org/organizations/greenpeace-org/issues/452

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [umbriel](https://www-dev.greenpeace.org/test-umbriel) | [admin](https://www-dev.greenpeace.org/test-umbriel/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-umbriel/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-umbriel) | [composer-local.json](https://github.com/greenpeace/planet4-test-umbriel/blob/main/composer-local.json)

:watch: 2024.10.28 11:45:47

File: taxonomy.php [hunk 1]
```diff
@@ -16,7 +16,7 @@
 $taxonomy = get_queried_object();
```
Comment (GP-Dan-Tovbein) [inline taxonomy.php:16]: Thinking loud.. have you tried the above code instead?
```suggestion
$taxonomy = get_queried_object()->taxonmy;
```

File: src/DevReport.php [hunk 2]
```diff
@@ -69,25 +69,44 @@ public function admin_page_display(): void
         }
 
         foreach ($gp_packages as $gp_package) {
-            $url = $gp_package[2]['url'];
-            if ('.git' === substr($url, -4)) {
-                $url = substr($url, 0, -4);
+            $pack0 = isset($gp_package[0]) ? $gp_package[0] : null;
```
[the remark this answers is not in the document]
Comment (mardelnet) [reply src/DevReport.php:72]: Thanks @GP-Dan-Tovbein
By using your approach, if `$gp_package[0]` is not set, `$pack0` will be `null` but we will still get a notice about an undefined index in Sentry.

File: taxonomy.php (hunk 1 above)
Comment (mardelnet) [reply taxonomy.php:16]: I think I don't understand this suggestion.
If the value of `$taxonomy` is `get_queried_object()->taxonomy` then `$taxonomy` becomes a string and we cannot access `$taxonomy->name` or `$taxonomy->description`

Comment (GP-Dan-Tovbein) [reply taxonomy.php:16]: You are correct! I saw only this part https://github.com/greenpeace/planet4-master-theme/pull/2425/files#diff-0a1d6d9a542b4540027ce80be62d507972d828c8b530869ac2743b056a2664d4L18

**Comment by mleray** [comment]
@mardelnet the e2e tests are failing because somehow the "Campaign" style of the Covers block isn't available anymore, can you look into that?

**Comment by mardelnet** [comment]
> @mardelnet the e2e tests are failing because somehow the "Campaign" style of the Covers block isn't available on the instance, can you look into that?

Fixed @mleray

File: taxonomy.php [hunk 3]
```diff
@@ -16,7 +16,7 @@
...
 
 $context['taxonomy'] = $taxonomy;
-$context['wp_title'] = $taxonomy->name;
-$context['og_description'] = $taxonomy->description;
+$context['wp_title'] = $taxonomy ? $taxonomy->name : '';
```
Comment (mleray) [inline taxonomy.php:19]: Should we even do anything if the `taxonomy` is undefined? 🤔

Comment (mleray) [reply taxonomy.php:19]: It's weird that this template would be called then

Comment (mardelnet) [reply taxonomy.php:19]: I know and agree with you.
However, as we cannot see the traces of these errors on Sentry ([here](https://sentry.greenpeace.org/organizations/greenpeace-org/issues/202/?project=2) and [here](https://sentry.greenpeace.org/organizations/greenpeace-org/issues/452/?project=2)) I just focused on fixing the reported warning and avoiding additional changes with unexpected consequences.
What do you think?

Comment (mleray) [reply taxonomy.php:19]: Yeah I think that makes sense!

File: src/CustomTaxonomy.php [hunk 4]
```diff
@@ -91,6 +91,11 @@ public function create_taxonomy_metabox_markup(WP_Post $post): void
         if (-1 === $current_type) {
             // Assign default p4-pagetype for new POST.
             $planet4_default_post_type = $this->get_planet4_default_post_type();
+
+            if (!$planet4_default_post_type) {
```
Comment (mleray) [inline src/CustomTaxonomy.php:95]: I'm not fully sure where this is used, but if we `return;` here the rest of the function won't be executed right? So the `select` component [here](https://github.com/greenpeace/planet4-master-theme/blob/b8631d4a7deddfe8efc5c81dc3fb8c8db9e8d677/src/CustomTaxonomy.php#L104-L110) won't be added?

**Review by mleray — APPROVED** [review]
Looks good to me 👍

**Pull request merged**